src/components/diff-view.jsx

```diff
@@ -24,7 +26,7 @@ import SideBySideRenderedDiff from './side-by-side-rendered-diff';
 export default class DiffView extends React.Component {
   constructor (props) {
     super(props);
-    this.state = {diff: null};
+    this.state = {diff: null, loadDiffType: null};
```

Comment:
I think something like `loadDiffType` is only really needed if we might be displaying one diff type while loading another. Since we are now showing a loading view during this time instead (definitely an improvement!), I don’t think we need `loadDiffType`. When starting to load a new diff, we should just set `diff` back to `null`.

Reply:
Hmmmm, actually, I take this back (sorta). We probably need a loading flag so that we can handle loading errors (obviously we don’t support that yet, but we should), e.g:

```js
this.state = {diff: null, loading: false, error: null};
```

Though I suppose we could just stuff the error into `diff` and have the `render` be smart enough to figure out whether `diff` is an actual diff or an error object.

Reply:
Makes sense. Made all the changes to get rid of loadDiffType

Reply:
Lol, I just saw the second comment. I was thinking of adding a loading flag too! But, everything worked without it, so I didn't. I'm going out of town this weekend. Let's merge and iterate if we can. I'll open an issue, so we don't forget.